soft_999_Jenish_Pate_fw_103

```diff
@@ -0,0 +1,73 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct CounterStorage {
+  uint16_t potentiometer_data;   // Stores data
+  AdcChannel address;            // Stores channel
+} CounterStorage;
+
+ void prv_button_interrupt_handler(const Gpiovoid *empty) {    
```

Comment:
This isn't the right type for a gpio interrupt callback, please call it something other than empty as well